### DISCUSSION TITLE
Dry signal only?

### DISCUSSION THREAD

**Discussion by atlogik** [jamulussoftware/jamulus#3375] [Technical Support & Help] [answered]
I use an IK Multimedia Axe I/O interface and have several different plugins I can use for amps, etc but am currently trying to use Amplitube.

I cannot get audio to route from my Amplitube app to Jamulus.

I've installed ASIO4ALL and have selected that in my amplitube app for both in/out and have audio as normal.
I've installed ASIO Bridge and enabled that
I've installed Jack Audio in order to try to re-route audio and have the ASIO4ALL driver selected as the interface
I've set Jamulus to use Jackaudio and tried to use the graph to route my signal chain accordingly.

I get either no audio at all in Jamulus, or I get a dry signal with no modeling applied from my app.

How are you guys making this work in Windows? I just want to play guitar with my amp modeler and use that as my input for Jamulus.

Are most of you guys using a mic and cabinet for guitar? 

Any help is greatly appreciated.

**Comment by dcorson-ticino-com** [comment]
Does your interface work correctly with just ASIO4ALL and Jamulus, in the simplest configuration?
I find ASIO4ALL to be highly problematic.

**Reply by atlogik** [reply]
Yep, I can choose ASIO4ALL or my AXEIO ASIO driver and get the dry input signal from my guitar just fine, there's just no overdrive or effects from my amp plugins. So I guess the input is working as intended, the part im confused about is how to get the signal from my amp sim software OUT and back into Jamulus. I also have Ableton as a DAW, but couldn't figure out how to route that either.

**Comment by ann0see** [comment]
Maybe you need to adapt https://jamulus.io/kb/2022/01/26/Routing-JACK-Windows.html - hopefully it shows your error. 

Take a screenshot of the routing configuration + all the related programs.

Does your DAW/Amplitube support JACK natively? If so, you can use the JACK version of Jamulus directly.

**Comment by atlogik** [comment, marked as answer]
I was able to get this working using Reaper and re-routing my return audio through Jamulus.

This guy has a great walkthrough of how to accomplish this using VST's and an audio interface without the need for JACK, Virtual Cable & ASIO4ALL.

https://www.youtube.com/watch?v=hRWIJqgRUXc&ab_channel=TekPandaUK

**Comment by uazure** [comment]
It'd be nice to have some general solution at least for windows (ASIO). Though I believe it can work with Reaper fine enough, I'd be happy to have some better solution that allows to use any ASIO-supporting software.

In scope of jamulus development it could be an optional virtual ASIO device that is installed to the operating system.
Jamulus continues using the real hardware ASIO driver (e.g., MOTU, RME, Focusrite) directly.
A separate lightweight virtual ASIO device (e.g., “Jamulus Link ASIO”) is exposed to the system. (might be opt-in on jamulus setup)


In this mode jamulus would get audio input from real device asio driver and send it to a “Jamulus Link ASIO”. And it would read audio from “Jamulus Link ASIO” output and send it to the server. Received audio from server will be played back on real device, and also might be pushed to “Jamulus Link ASIO” (some additional channel) so that DAW could record network stream easily.

If I'm not mistaken, “Jamulus Link ASIO” wouldn't be just a single asio driver, it might be more complicated that that. It might be two separate virtual asio devices each with some common module behind it. But that's an implementation detail

And it might have been a standalone solution, which is abstracted from jamulus. But that's a different story. Maybe there are some windows c++ developers here who could do that? Or maybe we might collaborate on this somehow?
